sources/packages/web/src/components/aest/StudentApplications.vue

```diff
@@ -0,0 +1,290 @@
+          {{ dateString(slotProps.data.studyEndPeriod) }}
+        </span>
+      </template></Column
+    >
+    <Column
+      :field="StudentApplicationFields.Status"
+      header="Status"
+      sortable="true"
+    >
+      <template #body="slotProps">
+        <Status :statusValue="slotProps.data.status" />
+      </template>
+    </Column>
+    <Column :field="StudentApplicationFields.Actions" header="Actions">
+      <template #body="slotProps">
+        <span v-if="clientType === ClientIdType.Student">
+          <span
+            v-if="
+              !(
+                slotProps.data.status === ApplicationStatus.cancelled ||
+                slotProps.data.status === ApplicationStatus.completed
+              )
+            "
+          >
+            <v-btn :disabled="hasRestriction" plain>
+              <v-icon
+                size="25"
+                v-tooltip="'Click To Edit this Application'"
+                @click="
+                  slotProps.data.status !== ApplicationStatus.draft
+                    ? confirmEditApplication(slotProps.data.id)
+                    : editApplicaion(slotProps.data.id)
+                "
+                >mdi-pencil</v-icon
+              ></v-btn
+            >
+            <v-btn :disabled="hasRestriction" plain>
+              <v-icon
+                size="25"
+                v-tooltip="'Click To Cancel this Application'"
+                @click="openConfirmCancel(slotProps.data.id)"
+                >mdi-trash-can-outline</v-icon
+              >
+            </v-btn>
+          </span>
+        </span>
+        <span v-if="clientType === ClientIdType.AEST">
+          <v-btn outlined>View</v-btn>
+        </span>
+      </template>
+    </Column>
+  </DataTable>
+  <CancelApplication
+    :showModal="showModal"
+    :applicationId="selectedApplicationId"
+    @showHideCancelApplication="showHideCancelApplication"
+    @reloadData="reloadApplication"
+  />
+  <ConfirmEditApplication ref="editApplicationModal" />
+</template>
+
+<script lang="ts">
+import { onMounted, ref } from "vue";
+import {
+  StudentDetail,
+  ApplicationStatus,
+  StudentApplicationAndCount,
+  DEFAULT_PAGE_LIMIT,
+  DEFAULT_PAGE_NUMBER,
+  DataTableSortOrder,
+  PAGINATION_LIST,
+  StudentApplicationFields,
+  ProgramYearOfApplicationDto,
+  ClientIdType,
+} from "@/types";
+import { ApplicationService } from "@/services/ApplicationService";
+import { StudentService } from "@/services/StudentService";
+import { useFormatters, ModalDialog } from "@/composables";
+import Status from "@/views/student/ApplicationStatus.vue";
+import { useRouter } from "vue-router";
+import { StudentRoutesConst } from "@/constants/routes/RouteConstants";
+import ConfirmEditApplication from "@/components/students/modals/ConfirmEditApplication.vue";
+import CancelApplication from "@/components/students/modals/CancelApplicationModal.vue";
+import Tooltip from "primevue/tooltip";
```

Comment:
why didn't we import tooltip at main.ts level?

Reply:
old code moved to main.ts